care/emr/api/viewsets/specimen.py

```diff
@@ -0,0 +1,203 @@
+from datetime import UTC, datetime
+
+from django.db import models
+from django.db.models import Case, CharField, Q, Value, When
+from django_filters import ChoiceFilter, FilterSet, OrderingFilter, UUIDFilter
+from django_filters.rest_framework import DjangoFilterBackend
+from drf_spectacular.utils import extend_schema, extend_schema_view
+from rest_framework.decorators import action
+from rest_framework.generics import get_object_or_404
+from rest_framework.response import Response
+
+from care.emr.api.viewsets.base import EMRModelViewSet
+from care.emr.fhir.schema.base import DateTime
+from care.emr.models.specimen import Specimen
+from care.emr.resources.specimen.spec import (
+    SpecimenCollectRequest,
+    SpecimenCreateSpec,
+    SpecimenListSpec,
+    SpecimenProcessRequest,
+    SpecimenReceiveAtLabRequest,
+    SpecimenRetrieveSpec,
+    SpecimenSendToLabRequest,
+    SpecimenUpdateSpec,
+    StatusChoices,
+)
+
+
+class PhaseChoices(models.TextChoices):
+    ORDERED = "ordered", "Ordered"
+    COLLECTED = "collected", "Collected"
+    SENT = "sent", "Sent"
+    RECEIVED = "received", "Received"
+    IN_PROCESS = "in_process", "In Process"
+
+
+class SpecimenFilters(FilterSet):
+    request = UUIDFilter(field_name="request__external_id")
+    encounter = UUIDFilter(field_name="request__encounter__external_id")
+    phase = ChoiceFilter(choices=PhaseChoices.choices, method="filter_phase")
+
+    ordering = OrderingFilter(
+        fields=(
+            ("created_date", "created_date"),
+            ("modified_date", "modified_date"),
+        )
+    )
+
+    def filter_phase(self, queryset, name, value):
+        return queryset.annotate(
```

Comment:
Maintain it as a calculated attribute or a filter, we will not do any annotation